core/blockchain/abci/abci.go

```diff
@@ -194,6 +195,161 @@ func AddCommonDeliverTxEvents(resp types.ResponseDeliverTx, tx Tx) types.Respons
 	return resp
 }
 
+const (
+	TxTypeStakeDeposited                 = "StakeDeposited"
+	TxTypeStakeRemoved                   = "StakeRemoved"
+	TxTypeBuiltinAssetDeposit            = "BuiltinAssetDeposit"
+	TxTypeBuiltinAssetWithdrawal         = "BuiltinAssetWithdrawal"
+	TxTypeERC20Deposit                   = "ERC20Deposit"
+	TxTypeERC20Withdrawal                = "ERC20Withdrawal"
+	TxTypeTransfer                       = "Transfer"
+	TxTypeOrderSubmission                = "OrderSubmission"
+	TxTypeOrderCancellation              = "OrderCancellation"
+	TxTypeOrderAmendment                 = "OrderAmendment"
+	TxTypeVoteSubmission                 = "VoteSubmission"
+	TxTypeWithdrawSubmission             = "WithdrawSubmission"
+	TxTypeLiquidityProvisionSubmission   = "LiquidityProvisionSubmission"
+	TxTypeLiquidityProvisionCancellation = "LiquidityProvisionCancellation"
+	TxTypeProposalSubmission             = "ProposalSubmission"
+	TxTypeAnnounceNode                   = "AnnounceNode"
+	TxTypeNodeVote                       = "NodeVote"
+	TxTypeNodeSignature                  = "NodeSignature"
+	TxTypeOracleDataSubmission           = "OracleDataSubmission"
+	TxTypeDelegateSubmission             = "DelegateSubmission"
+	TxTypeUndelegateSubmission           = "UndelegateSubmission"
+	TxTypeKeyRotateSubmission            = "KeyRotateSubmission"
+	TxTypeStateVariableProposal          = "StateVariableProposal"
+	TxTypeCancelTransfer                 = "CancelTransfer"
+	// TxTypeValidatorHeartbeat                 = "ValidatorHeartbeat".
+	TxTypeEthereumKeyRotateSubmission = "EthereumKeyRotateSubmission"
+	TxTypeProtocolUpgradeProposal     = "ProtocolUpgradeProposal"
+	TxTypeIssueSignatures             = "IssueSignatures"
+	TxTypeBatchMarketInstructions     = "BatchMarketInstructions"
+	TxTypeERC20MultisigSignerAdded    = "ERC20MultisigSignerAdded"
+	TxTypeERC20MultisigSignerRemoved  = "ERC20MultisigSignerRemoved"
+)
+
+type TxType struct {
+	Type     string
+	Sender   string
```

Comment:
Seems like Sender here, seeing how it's used my be a duplicate to the `submitter` label which is already added somewhere for every transaction isn't?

Reply:
@jeremyletang I'm fine merging those two if we can assume that the submitter will always be the same as the sender (for example in transfers). However, there is an exception in `GetErc20Multisig`. It wouldn't make sense to have an attribute for every single field, so we settle for the 3 we have now: `type`, `submitter`, `sender` and `receiver`. I'm open to suggestions here, I know this is not ideal